gen3utils/gitops/gitops_validator.py

```diff
@@ -0,0 +1,385 @@
+
+    return errors
+
+
+def validate_explorerConfig(gitops, type_prop_map, errors):
+    exploreConfig = gitops.get("explorerConfig") or gitops.get("dataExplorerConfig")
+    # if explorerConfig exists, ignores (data/files)explorerConfig
+    if exploreConfig and type(exploreConfig) == list:
+        for config in exploreConfig:
+            errors = _validate_explorerConfig_helper(config, type_prop_map, errors)
+    else:
+        errors = _validate_explorerConfig_helper(exploreConfig, type_prop_map, errors)
+        file_exp_config = gitops.get("fileExplorerConfig")
+        if file_exp_config:
+            errors = _validate_explorerConfig_helper(
+                file_exp_config, type_prop_map, errors
+            )
+
+    return errors
+
+
+def validate_against_etl(gitops, mapping_file):
+    """
+    Validates gitops.json configuration against an etlMapping
+    Args:
+        gitops (dict): gitops.json config
+        mapping_file (str): path to etlMapping file
+
+    Returns:
+        Error: Returns a list of any errors encountered.
+
+    """
+    with open(mapping_file) as f:
+        mappings = yaml.safe_load(f)
+    mapping = mappings.get("mappings")
+    type_prop_map = map_all_ES_index_props(mapping)
+    errors = validate_explorerConfig(gitops, type_prop_map, [])
+    studyviewer = gitops.get("studyViewerConfig")
+    if studyviewer:
+        errors = validate_studyViewerConfig(studyviewer, type_prop_map, errors)
+
+    return errors
+
+
+def map_all_ES_index_props(mapping):
+    all_prop_map = {}
+    for index in mapping:
+        index_props = []
+        props = index.get("props")
+        index_props.extend(_extract_props(props))
+        agg_props = index.get("aggregated_props")
+        index_props.extend(_extract_props(agg_props))
+        join_props = index.get("joining_props", [])
+        for indx in join_props:
+            index_props.extend(_extract_props(index.get("props")))
+        inject_props = index.get("injecting_props", {})
+        for node, props in inject_props.items():
+            index_props.extend(_extract_props(props.get("props")))
+        flat_props = index.get("flatten_props", [])
+        for node in flat_props:
+            index_props.extend(_extract_props(node.get("props")))
+        parent_props = index.get("parent_props")
+        if parent_props:
+            for prop in parent_props:
+                item = prop.get("path")
+                # to extract values from a string like "subjects[subject_id:id,project_id]"
+                [_, str_props] = (
+                    list(filter(None, re.split(r"[\[\]]", item)))
+                    if "[" in item
+                    else [item, None]
+                )
+                if str_props is not None:
+                    props = str_props.split(",")
+                    index_props.extend(
+                        [
+                            p.split(":")[0].strip() if p.find(":") != -1 else p.strip()
```

Comment:
let's not strip here - we actually ran into an issue recently, the portal crashed because the ETL mapping had an extra space: https://github.com/uc-cdis/gitops-qa/pull/1243/files